### ISSUE TITLE
Feature request / Refactor IExceptionHandler

### ISSUE THREAD

**Issue by yamaoto** [CynosuraPlatform/Cynosura#1]
# The problem
* Very hard to override default logic like in ServiceExceptionHandler
* We have to write separate handlers for each exception type
* More often we need to slightly change the error model than to change the whole response format 

# Implementation

1. Change IExceptionHandler
  public interface IExceptionHandler
  {
// add priority property for adding more highly prioritized handlers
     int Priority { get; }

      ~~Type ExceptionType { get; }~~
// filter by exception types or exception values instead of ExceptionType 
      bool Validate(Type exceptionType, Exception exception);

      ~~ObjectResult~~ ExceptionHandleResult HandleException(Exception exception);
  }

2. Add class ExceptionHandleResult
```
  public class ExceptionHandleResult
  {
    public ExceptionHandleResult()
    {
    }

    public ExceptionHandleResult(object model)
    {
      this.Model = model;
    }

    // model for error representation
    public object Model { get; set; }

   // override default IActionResilt presenter
    public Func<ExceptionHandleResult, ExceptionContext, IActionResult> OverrideResult { get; set; }
  }
```

3. In ApiExceptionFilterAttribute response by default with JsonResult and HTTP 500

4. Move ServiceExceptionHandler to template

**Comment by bacr** [comment]
1. `bool Validate(Type exceptionType, Exception exception);` - probably `bool Validate(Exception exception);` is enough?
2. Why do you need OverrideResult? How it will be used?

**Comment by yamaoto** [comment]
1. For cases when you need to check some values from exception like code
2. In ApiExceptionFilterAttribute we can set result with ObjectResult by default. With OverrideResult we can set custom HTTP Code

**Comment by bacr** [comment]
You can implement it with some tests and I will see. Otherwise it is not clear how it will work.

**Comment by bacr** [comment]
Fixed in 3737c9437a97eff2381c9b1db189bc5b23a51792
